locations/jclouds/src/test/java/org/apache/brooklyn/location/jclouds/BasicJcloudsLocationCustomizerTest.java

```diff
@@ -0,0 +1,81 @@
+import org.apache.brooklyn.core.entity.BrooklynConfigKeys;
+import org.apache.brooklyn.core.test.BrooklynAppUnitTestSupport;
+import org.apache.brooklyn.core.test.entity.TestEntity;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+
+public class BasicJcloudsLocationCustomizerTest extends BrooklynAppUnitTestSupport {
+
+    private BasicJcloudsLocationCustomizer testCustomizer;
+
+    @BeforeMethod
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        testCustomizer = new BasicJcloudsLocationCustomizer();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testCustomiserIncluded() {
+
+        TestEntity entity = app.createAndManageChild(EntitySpec.create(TestEntity.class)
+                .addInitializer(testCustomizer));
+        Object object = entity.config().get(BrooklynConfigKeys.PROVISIONING_PROPERTIES.subKey(
+                JcloudsLocationConfig.JCLOUDS_LOCATION_CUSTOMIZERS.getName()));
+        assertNotNull(object, "expected value for customizers in " + entity.config().get(BrooklynConfigKeys.PROVISIONING_PROPERTIES));
+        assertTrue(object instanceof Collection, "expected collection, got " + object.getClass());
+        Collection<JcloudsLocationCustomizer> customizers = (Collection<JcloudsLocationCustomizer>) object;
+        assertEquals(customizers.size(), 1, "expected single customizer in " + Iterables.toString(customizers));
+        assertTrue(customizers.contains(testCustomizer), "expected to find testCustomizer in " + Iterables.toString(customizers));
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testCustomizersMerged() {
+        TestEntity entity = app.createAndManageChild(EntitySpec.create(TestEntity.class)
+                .addInitializer(testCustomizer)
+                .configure(
+                        BrooklynConfigKeys.PROVISIONING_PROPERTIES.subKey(JcloudsLocationConfig.JCLOUDS_LOCATION_CUSTOMIZERS.getName()),
+                        ImmutableList.of(new BasicJcloudsLocationCustomizer())));
+        Object object = entity.config().get(BrooklynConfigKeys.PROVISIONING_PROPERTIES.subKey(
+                JcloudsLocationConfig.JCLOUDS_LOCATION_CUSTOMIZERS.getName()));
+        assertNotNull(object, "expected value for customizers in " + entity.config().get(BrooklynConfigKeys.PROVISIONING_PROPERTIES));
+        assertTrue(object instanceof Collection, "expected collection, got " + object.getClass());
+        Collection<JcloudsLocationCustomizer> customizers = (Collection<JcloudsLocationCustomizer>) object;
+        assertEquals(customizers.size(), 2, "expected two customizers in " + Iterables.toString(customizers));
+        assertTrue(customizers.contains(testCustomizer), "expected to find testCustomizer in " + Iterables.toString(customizers));
+    }
+
```

Comment:
Thought that there's an `addInitializer` method on the entity, but that's not the case.
To add an initializer to an existing entity create the initializer and call `apply` on the entity.